examples/hosts/iframe-host/src/containerProxy.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as Comlink from "comlink";
+import { IContainer } from "@fluidframework/container-definitions";
+import { IRequest, IResponse } from "@fluidframework/core-interfaces";
+import { IQuorum } from "@fluidframework/protocol-definitions";
+
+export interface IContainerProxy {
+    attach(request: IRequest): Promise<void>;
+    request(request: IRequest): Promise<IResponse>;
+    getQuorum(): Promise<IQuorum>;
```

Comment:
>getQuorum(): Promise<IQuorum>; [](start = 4, length = 30)

originally we didn't need to proxy the quorum, as that part of the Container could live outside the iframe, as it didn't require any untrusted code, only the container code. i see why things like attach make that more complicated, as that will need to be proxied. we'll probably want a hybrid approach eventually. my intuition is that we'll need a full proxy for the detached case, but the load case won't need as much/any, as we can just use the Container

Reply:
we used the ProxyRuntime to get the full container on the outside. i'd like to keep support for that

---
In reply to: [530573659](https://github.com/microsoft/FluidFramework/pull/4404#discussion_r530573659) [](ancestors = 530573659)